### PR TITLE
Remove sfv1 to sfv2.1 migration code

### PR DESCRIPTION
**Pull request by muralisc**
1. BnR status poller migration code
2. Last operation lock
3. Create deployment resource in state succeeded
4. Add error handling to BackupService.

### CODE REVIEW

**Comment by amitmalav** [comment]
@muralisc Can you please make change at these places also to remove migration code for lock in `lastOperation` call
https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/broker/lib/middleware.js#L47
https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/api-controllers/routes/broker/v2.js#L41

File: api-controllers/routes/broker/v2.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ instanceRouter.route('/')
   .delete([middleware.validateRequest(), controller.handleWithResourceLocking('deleteInstance', CONST.OPERATION_TYPE.DELETE)])
   .all(commonMiddleware.methodNotAllowed(['PUT', 'PATCH', 'DELETE']));
 instanceRouter.route('/last_operation')
-  .get([middleware.lock(undefined, true), controller.handler('getLastInstanceOperation')]) //passing undefined as last operation operationType is part of the req
+  .get([controller.handler('getLastInstanceOperation')]) //passing undefined as last operation operationType is part of the req
```
Comment (amitmalav) [inline api-controllers/routes/broker/v2.js]: Can remove array.
`.get([controller.handler('getLastInstanceOperation')]) ` => `.get(controller.handler('getLastInstanceOperation'))`

File: managers/backup-manager/BackupService.js [hunk 2]
```diff
@@ -372,6 +374,17 @@ class BackupService extends BaseDirectorService {
...
         }
+      })
+      .catch(e => {
+        return eventmesh.apiServerClient.updateResource({
+          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+          resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+          resourceId: abortOptions.guid,
+          status: {
+            state: CONST.APISERVER.RESOURCE_STATE.FAILED,
+            error: e
```
Comment (amitmalav) [inline managers/backup-manager/BackupService.js]: If error object is something other than our custom error, it will be set as empty object as we saw.

Comment (muralisc) [reply managers/backup-manager/BackupService.js]: Building custom error now.

File: api-controllers/routes/broker/v2.js (hunk 1 above)
Comment (muralisc) [reply api-controllers/routes/broker/v2.js]: Removed.

**Comment by muralisc** [comment]
Dev pipeline # 329 is being run. Waiting for it to complete

**Comment by muralisc** [comment]
Pipeline is green , can be merged

**Pull request merged**